### ISSUE TITLE
mbe.modelica.university appears to be down

### ISSUE THREAD

**Issue by mattdailis** [mtiller/ModelicaBook#467]
Hi there! I just wanted to let you know that all of the links to the book preview seem to be non-functional at the moment:

<img width="844" alt="Screenshot 2023-12-22 at 9 00 35 AM" src="https://github.com/mtiller/ModelicaBook/assets/1189602/ce56e8c1-992a-4267-8771-7cb292936e5e">

I'd love to read your book-in-progress, but I'm having trouble finding it! Looks like I might be able to scrape something from google's cache...

I attempted to build this repository, but ran into trouble there as well. I used `pip` to install `jinja2` and `dvc`, but then got stuck on `omc` - looks like recent versions of Modelica don't explicitly support Macs. I will attempt to get it running via Docker, and see if I can build this repository in a docker container. I am new to Modelica, and would love to use your book as a starting point.

EDIT: Upon a closer reading of the README, I see that you do recommend using Docker. That said, I can't quite glean what steps I should take from this description - in my first attempt, I skipped straight to the `git submodule` and `make` commands. Should I be running those from a Docker context? What base image would you recommend using?

> Sphinx is completely portable so in theory, it should be possible to run Sphinx under nearly any operating system.
> 
> Although previous versions required a bunch of tools to build the book, the current version only requires that you have Node and Docker installed. I highly recommend the [Docker for Mac](https://www.docker.com/docker-mac) and [Docker for Windows](https://www.docker.com/docker-windows) flavors if you are on macOS or Windows, respectively.
> 
> Recently, I had some success using the Remote - Containers extension in Visual Studio Code along with the mtiller/flat-book-builder image mentioned below. If you open this folder in Visual Studio Code and you have the Remote - Containers extension installed, you should get an option to re-open VS Code in the container. Doing so, you should be setup to build. As such, I've bundled the .devcontainer directory that include the configuration necessary to do the builds in containers. This could also be useful when building on Windows (if you have VS Code and the Remote - Containers extension installed).
> 
> In order to build this, the first thing you need to do is ensure all submodules have been checked out, i.e.,
> 
> $ git submodule init
> $ git submodule update
> 
> There is now a root level Makefile. You should be able to simply run:
> 
> make all
> 
> ...which should build the book (if I'm forgetting something, send me a pull request for this README adding any further instructions). The source will be generated in ./text/build/dirhtml.

**Comment by mtiller** [comment]
Sorry about that.  The auto-renewal on the domain failed because the credit card I had on file had expired.  I've requested renewal of the domain and hopefully everything will sort it self out in the coming hours.

As for building the book, I am using a Mac so it definitely will build on a Mac.  I did some work on this about three months ago and revamped much of it.  I don't recall off the top of my head what configuration I used for Docker on my Mac but I'm pretty sure it was possible.  I might have been using Rosetta and perhaps specifying a Linux on AMD64 somewhere in the configuration.  I'm afraid I don't recall off the top of my head and with the holidays right now, I probably won't have time to figure this out during the holidays.  Sorry that I didn't leave enough "breadcrumb trails" on this.  I usually try to write things up so that even I don't forget it.
